src/main/java/net/sf/jabref/model/database/BibDatabase.java

```diff
@@ -413,11 +439,14 @@ private String resolveString(String label, Set<String> usedIds) {
                 }
                 // If not, log this string's ID now.
                 usedIds.add(string.getId());
```

Comment:
Here you get the id of the string and above in getUsedStrings you get the string from the id.
Thus I would propose to change `Set<String> usedIds` to `Set<BibString> usedStrings`.